src/integrations/cleanup-integration.php

```diff
@@ -41,10 +42,69 @@ public function cleanup_obsolete_indexables( $object_type, $object_sub_type ) {
 		$number_of_deletions = $this->clean_indexables_with_object_type( $object_type, $object_sub_type, 1000 );
 
 		if ( empty( $number_of_deletions ) ) {
-			$this->unschedule_cron();
+			$this->unschedule_cron( 'wpseo_cleanup_indexables' );
```

Comment:
this would unschedule the cron even if we decide to use the cleanup/housekeeping action for something entirely different in the future. I don't think that's the intent of the ticket.

Reply:
I think that it _should_ unschedule the cron, as this is the intent of the ticket.

For the future, we can always use the new action functions (`clean_indexables_with_object_type` & `cleanup_orphaned_from_table`) but the wrapper functions that are being called by the cron hooks, should be specific to this task at hand (so, the `cleanup_obsolete_indexables` and `cleanup_orphaned_indexables` ones).

Also that's a point about not renaming the cron jobs too. The cron jobs should have a name that's specific to the task at hand. We still have code to extend/use in the future, but only the action functions (`clean_indexables_with_object_type` & `cleanup_orphaned_from_table`).

IMO 😃